react/components/SearchBar.js

```diff
@@ -1,52 +1,69 @@
 import React from 'react'
-import PropTypes from 'prop-types'
+import { func, bool, string } from 'prop-types'
 import { FormattedMessage } from 'react-intl'
 import { Adopt } from 'react-adopt'
-import { ExtensionPoint } from 'vtex.render-runtime'
+import { ExtensionPoint, withRuntimeContext } from 'vtex.render-runtime'
 import { Button } from 'vtex.styleguide'
-
+import classNames from 'classnames'
 import header from '../store-header.css'
+import { CONSTANTS } from './Helpers'
+
+const SearchBar = ({
+  autoFocus,
+  onCancel,
+  iconClasses,
+  runtime: { hints: { mobile, desktop } } 
+}) => {
+
+  const searchBarClassNames = classNames(`${header.topMenuSearchBar} flex pa2-m flex-grow-1`, {
+    'justify-between': mobile,
+    'justify-center': desktop
+  })
+
+  const cancelClassNames = classNames(`${iconClasses} ttl`)
 
-const SearchBar = ({ compactMode, autoFocus, onCancel }) => {
   return (
-    <React.Fragment>
-      <div className={`${header.topMenuSearchBar} flex pa2-m flex-grow-1 justify-center`}>
-        <div className="w-100 mw7">
-          <Adopt
-            mapper={{
-              placeholder: <FormattedMessage id='header.search-placeholder' />,
-              emptyPlaceholder: <FormattedMessage id='header.search-emptyPlaceholder' />,
-            }}
-          >
-            {({ placeholder, emptyPlaceholder }) => (
-              <ExtensionPoint
-                id="search-bar"
-                placeholder={placeholder}
-                emptyPlaceholder={emptyPlaceholder}
-                compactMode={compactMode}
-                autoFocus={autoFocus}
-              />
-            )}
-          </Adopt>
-        </div>
+    <div className={searchBarClassNames}>
+      <div className="w-75">
+        <Adopt
+          mapper={{
+            placeholder: <FormattedMessage id='header.search-placeholder' />,
+            emptyPlaceholder: <FormattedMessage id='header.search-emptyPlaceholder' />,
+          }}
+        >
+          {({ placeholder, emptyPlaceholder }) => (
+            <ExtensionPoint
+              id="search-bar"
+              placeholder={placeholder}
+              emptyPlaceholder={emptyPlaceholder}
+              autoFocus={autoFocus}
+              hasIconLeft={mobile}
+              iconClasses={iconClasses}
+            />
+          )}
+        </Adopt>
       </div>
-      {compactMode && (
-        <Button size="small" variation="tertiary" onClick={onCancel}>
-          <FormattedMessage id="header.search-cancel" />
-        </Button>
+      { mobile && (
+        <div className="w-25 pa2-m pt2-s">
+          <Button size="small" variation="tertiary" onClick={onCancel}>
+            <span className={cancelClassNames}><FormattedMessage id="header.search-cancel" /></span>
+          </Button>
+        </div>
       )}
-    </React.Fragment>
+    </div>
   )
 }
 
 SearchBar.propTypes = {
-  onCancel: PropTypes.func,
-  compactMode: PropTypes.bool,
-  autoFocus: PropTypes.bool,
+  onCancel: func,
```

Comment:
Add `runtime` ici... And also, there are some non-required props not describe on defaults

Reply:
✅